blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -74,29 +74,57 @@ namespace blueprint
             throw std::runtime_error(msg);
         }
 
+        //TODO: Keep safeEvalString and safeCompileFile or make EcmascripEngine member funcs
+        //      now that we have out errorHandle wrapper in EcmascriptEngine? Passing
+        //      callback here feels a bit ambiguous in regards to popping the error
+        //      off of the stack after invoking the error callback. Whose responsibility is it etc.
         template <typename T>
         static bool safeEvalString(duk_context* ctx, const juce::String& s, T&& callback)
         {
             if (duk_peval_string(ctx, s.toRawUTF8()) != 0)
             {
                 if (callback != nullptr)
                 {
-                    const juce::String trace = duk_safe_to_stacktrace(ctx, -1);
-                    const juce::String msg = duk_safe_to_string(ctx, -1);
+                    // Call the user provided error handler and indicate failure
+                    std::invoke(callback);
+                    return false;
+                }
+
+                duk_throw_raw(ctx);
+            }
 
-                    // Call the user provided error handler
-                    std::invoke(callback, msg, trace);
+            // Indicate success
+            return true;
+        }
 
-                    // Indicate failure
+        template<typename T>
+        static bool safeCompileFile(duk_context* ctx, const juce::File& file, T&& callback)
+        {
+            // Push the js filename to be compiled/evaluated
+            duk_push_string(ctx, file.getFileName().toRawUTF8());
+
+            if (duk_pcompile_string_filename(ctx, DUK_COMPILE_EVAL, file.loadFileAsString().toRawUTF8()) != 0)
+            {
+                if (callback != nullptr)
+                {
+                    // Call the user provided error handler.js and indicate failure
+                    std::invoke(callback);
                     return false;
                 }
 
-                 duk_throw_raw(ctx);
+                duk_throw_raw(ctx);
             }
 
             // Indicate success
             return true;
-        }
+       }
+
+       static void dumpContext(duk_context* ctx)
```

Comment:
I've been using this a lot whilst experimenting with this stuff so left it here in case you think it is worth using.